packages/contracts/src/contracts/current/protocol/Exchange/MixinSettlementProxy.sol

```diff
@@ -68,17 +69,20 @@ contract MixinSettlementProxy is
         )
     {
         makerTokenFilledAmount = getPartialAmount(takerTokenFilledAmount, order.takerTokenAmount, order.makerTokenAmount);
+        bytes32 orderHash = getOrderHash(order);
+
         require(
             TRANSFER_PROXY.transferFrom(
-                order.makerTokenAddress,
+                encodeMetadata(order.makerAssetProxyId, order.makerTokenAddress),
```

Comment:
I think we'll find that this will need to be made more generic as we start experimenting with different proxies for different standards.

Reply:
+1 I've removed this Enum and updated the encoder/decoder to be more flexible.